### PR TITLE
Full interruption support for GLPK

### LINKED ISSUE #897: `igraph_feedback_arc_set` should be interruptable
`igraph_feedback_arc_set` with method `IGRAPH_FAS_EXACT_IP` is not currently interruptible. Making this function interruptible is highly preferable because it can be hard to tell in advance whether it will finish in a reasonable amount of time for any given (directed) graph.

### PR DESCRIPTION
**Pull request by szhorvat**
This fixes #897 and #1382

Interruption does not work yet with the vendored GLPK, and I have not yet had time to figure out if this is because this old GLPK does not properly support it, or because changes were made to it. I have already undone some changes which would have prevented it from working but it still does not work. 

Nevertheless, comments are already welcome.

I am not totally happy with putting `igraph_abort()` into an internal `error.h`, as it is not easy to include in vendored sources.

### CODE REVIEW

**Comment by szhorvat** [comment]
OK, I found it. Now interruption works even with the vendored GLPK, but it's a bit slow. Unfortunately, interruption is tied to output printing, so there is an opportunity to interruption only as often as there is terminal output. I might have missed some places where output was disabled, which may be slowing down interruption. It is definitely slower with the vendored GLPK than an external GLPK 5.0.

**Comment by szhorvat** [comment]
This is not going to work as-is. It is leaking memory horribly on every interruption.

Interruption is not always interactive. A realistic use case is to compute the feedback arc set of a bunch of graph, with a time limit set on each (interruption when reaching the time limit). In fact, I have done precisely this with SageMath.

I need to understand what `glp_free_env()` does precisely and when it is safe to call it. If we do call it, we must avoid calling `glp_delete_prob()` after it, as it would lead to an abort/crash due to GLPK's overzealous error checking (see the checks in `glp_free`). Normally, `glp_delete_prob` would be called by `IGRAPH_FINALLY_FREE`.

**Comment by szhorvat** [comment]
Now we call `glp_free_env()`, no more memory leak on interruption.

File: src/internal/glpk_support.c [hunk 1]
```diff
@@ -32,6 +32,35 @@
...
+
+    if (igraph_i_interruption_handler) {
+        if (!igraph_i_glpk_error_info.is_interrupted && igraph_allow_interruption(NULL) != IGRAPH_SUCCESS) {
+            /* If an interruption has already occurred, do not set another error. */
+            igraph_i_glpk_error_info.is_interrupted = 1;
+            glp_error("GLPK was interrupted.");
+        } else {
+            /* Copy the error messages into a buffer for later reporting */
+            const size_t n = sizeof(igraph_i_glpk_error_info.msg) / sizeof(char) - 1;
+            while (*s != '\0' && igraph_i_glpk_error_info.msg_ptr < igraph_i_glpk_error_info.msg + n) {
```
Comment (ntamas) [inline src/internal/glpk_support.c:48]: This is essentially an `strncpy()`, isn't it?

Comment (szhorvat) [reply src/internal/glpk_support.c:48]: Yes, it is, but it seemed that due to the need of keeping track of the end of the string, and maintaining the null terminator, `strncpy` would not have simplified the code.

File: vendor/glpk/glpenv03.c [hunk 2]
```diff
@@ -45,9 +45,9 @@
 
 void glp_printf(const char *fmt, ...)
 {     va_list arg;
-      /* va_start(arg, fmt); */
-      /* xvprintf(fmt, arg); */
-      /* va_end(arg); */
+      va_start(arg, fmt);
+      xvprintf(fmt, arg);
```
Comment (ntamas) [inline vendor/glpk/glpenv03.c:49]: Would the R interface (or, well, more like the CRAN validation of the R interface) choke on the presence of a printf-like function?

**Comment by szhorvat** [comment]
I needed to do another hack and add `glp_at_error()` to our vendored GLPK to be able to determine which messages are error messages, and therefore should be collected for reporting.

Once we update the vendored GLPK, this hack will no longer be needed. New GLPK have this function.

**Comment by szhorvat** [comment]
I plan to merge this later today.

**Pull request merged**